Monster.cpp

```diff
@@ -21,13 +25,13 @@ double Monster::getAttackCoolDown() const {
 }
 
 void Monster::tamad(Monster* a) {
-	a->hp = a->hp - this->dmg;
+    if (this->dmg > a->def) a->hp = a->hp - (this->dmg - a->def);

```

Comment:
- itt a this-re szükség van?

Reply:
this re az essetek 99% nincs szükség, viszont én szeretem kitenni, mert plusz jelzés, hogy melyik osztályhoz tartozik, ha gond ki tudom szedni

Reply:
maradhat, akkor ezentúl nem szólok érte